src/Console/DuskCommand.php

```diff
@@ -58,12 +59,21 @@ public function handle()
 
         $this->purgeSourceLogs();
 
-        $options = array_slice($_SERVER['argv'], $this->option('without-tty') ? 3 : 2);
+        $options = collect(array_slice($_SERVER['argv'], 2))
+            ->filter(function ($option) {
+                return ! in_array($option, ['--without-tty', '--browse']);
+            })->values()->all();
 
         return $this->withDuskEnvironment(function () use ($options) {
+            $env = [];
+
+            if (! $this->option('browse')) {
+                $env['CI'] = true;
+            }
```

Comment:
Dusk uses headless by default so it may be more clear to name the environment variable with its `--browse` edge case description? Desktop test runs will also use headless mode (not in CI) until this new argument is used to analyze browser behaviour that the test failure screenshot doesn't capture.

Replacing this with the below code would help with packages keeping Laravel Dusk backwards capability:

```php
if ($this->option('browse')) {
    $env['DUSK_BROWSE'] = true;
}
```

```php
$options = (new ChromeOptions)->addArguments(collect([
    '--window-size=1920,1080',
])->unless(isset($_SERVER['DUSK_BROWSE']) || isset($_ENV['DUSK_BROWSE']), function ($items) {
    return $items->merge([
        '--disable-gpu',
        '--headless',
    ]);
})->all());
```

e.g., this `--browse` feature would be introduced in laravel/dusk 7.x so custom stubs must be updated to support it.

From my laravel-dusk-firefox package:

https://github.com/derekmd/laravel-dusk-firefox/blob/58d10303c580b120d0f70cb4a202d188c89a192b/stubs/DuskTestCase.stub#L56-L82

If `--headless` is only used when `$env['CI']` exists, there isn't a way to make the custom stub have backwards compatibility with laravel/dusk 6.x releases that don't set `$env['CI']`. There would need to be separate stubs for each major Dusk version.

Reply:
@crynobone any thoughts on comment above?

Reply:
Seem fine, just not sure if `DUSK_BROWSE` would be suitable. However both way still require stub TestCase to be updated in order to use `--browse`.

Reply:
Maybe `DUSK_HEADLESS_DISABLED` is more clear? Whatever the name, the main goal would be for `DuskTestCase` to look for the `--browse` option be enabled rather than the absence of that option.